cocotb/__init__.py

```diff
@@ -47,15 +48,11 @@
 # Things we want in the cocotb namespace
 from cocotb.decorators import test, coroutine, hook, function, external  # noqa: F401
 
-# Singleton scheduler instance
-# NB this cheekily ensures a singleton since we're replacing the reference
-# so that cocotb.scheduler gives you the singleton instance and not the
-# scheduler package
-
 from ._version import __version__
 
-# GPI logging instance
-if "COCOTB_SIM" in os.environ:
+
+def _setup_logging():
+    global log
 
     # sys.path normally includes "" (the current directory), but does not appear to when python is embedded.
     # Add it back because users expect to be able to import files in their test directory.
```

Comment:
This probably also belongs in `regression.py`

Reply:
If you don't want to handle this now, then perhaps just rename the function to `_setup_logging_warnings_and_path`